BuildScript/BuildTargets/Testing.targets

```diff
@@ -128,31 +124,37 @@
     <!-- Run tests -->
 
     <PropertyGroup>
-      <_testName>$(_testAssemblyFileName).$(ConfigurationID).$(_platform)</_testName>
-      <_testName Condition="'$(_databaseSystem)' != '$(NoDatabaseRequired)'">$(_testName).$(_databaseSystem)</_testName>
-      <_testName Condition="'$(_browser)' != '$(NoBrowserRequired)'">$(_testName).$(_browser)</_testName>
-      <_testResultFile>$(LogDirectory)$(_testName).xml</_testResultFile>
+      <_mergedTestCategoriesToExclude>$(TestCategoriesToExclude.Replace(';', ',')),$(_excludeCategoriesInTestConfiguration)</_mergedTestCategoriesToExclude>
+      <_mergedTestCategoriesToExclude>$(_mergedTestCategoriesToExclude.Trim(','))</_mergedTestCategoriesToExclude>
+
+      <_excludeCategoriesCallArgumentString></_excludeCategoriesCallArgumentString>
+      <_excludeCategoriesCallArgumentString Condition="'$(_mergedTestCategoriesToExclude)' != ''">/exclude=$(_mergedTestCategoriesToExclude)</_excludeCategoriesCallArgumentString>
     </PropertyGroup>
 
     <PropertyGroup>
-      <_dockerDisabled>$(DisableDocker)</_dockerDisabled>
-      <_dockerDisabled Condition="('$(_browser)' != '$(NoBrowserRequired)')">True</_dockerDisabled>
-      <_dockerDisabled Condition="('$(_executionRuntime)' == 'net-3.5')">True</_dockerDisabled>
+      <_mergedTestCategoriesToInclude>$(TestCategoriesToInclude.Replace(';', ',')),$(_includeCategoriesInTestConfiguration)</_mergedTestCategoriesToInclude>
+      <_mergedTestCategoriesToInclude>$(_mergedTestCategoriesToInclude.Trim(','))</_mergedTestCategoriesToInclude>
+
+      <_includeCategoriesCallArgumentString></_includeCategoriesCallArgumentString>
+      <_includeCategoriesCallArgumentString Condition="'$(_mergedTestCategoriesToInclude)' != ''">/include=$(_mergedTestCategoriesToInclude)</_includeCategoriesCallArgumentString>
     </PropertyGroup>
-    <Error Text="The property 'DotNetFrameworkRuntimeImageName' is not set." Condition="'$(DotNetFrameworkRuntimeImageName)' == '' And '(_dockerDisabled)' == 'False'" />
 
     <PropertyGroup>
-      <_nunitRunnerFileName>nunit-console.exe</_nunitRunnerFileName>
-      <_nunitRunnerFileName Condition="$(_use32Bit) == 'True'">nunit-console-x86.exe</_nunitRunnerFileName>
+      <_testName>$(_testAssemblyFileName).$(ConfigurationID).$(_platform).$(_executionRuntimeKey).$(_targetRuntime)</_testName>
+      <_testName Condition="'$(_databaseSystem)' != $(NoDB)">$(_testName).$(_databaseSystem)</_testName>
+      <_testName Condition="'$(_browser)' != $(NoBrowser)">$(_testName).$(_browser)</_testName>
+      <_testName Condition="'$(_mergedTestCategoriesToExclude)' !=''">$(_testName).Exclude=$(_mergedTestCategoriesToExclude.Replace(',', '-'))</_testName>
+      <_testName Condition="'$(_mergedTestCategoriesToInclude)' !=''">$(_testName).Include=$(_mergedTestCategoriesToInclude.Replace(',', '-'))</_testName>
```

Comment:
The `Replace` here is because our CI server apparently can't deal with commas in the filename (perhaps it's the CI server in conjunction with NUnit, nonetheless it works locally with commas). I wasn't sure if it's worth it to extract it inside a `<formattedTestCategoriesToExclude>` property (name TBD).

Reply:
I see. Sounds somehow fishy to me since there's no obvious difference aside from the MSBuild version perhaps. Maybe we have different values on teamcity and locally? I would like to understand the problem, not just have a workaround.

Reply:
The comma was somehow rendered as a newline, which broke the call. Since TeamCity calls MsBuild, which in turn calls NUnit, I brushed it off as a weird formatting thing that happens across 3 layers, especially since google gave me nothing on it. I could allocate additional time so for investigation if it's important to you

Reply:
Is it important. Good question. It's at least something that's not understood, meaning we will have issues with this in the future in some form or other. In theory, it could be the result of us doing somehting else wrong and this is just a workaround. It could also be that this is a well-defined behavior and we just can't use comma at this point. 

Should we analyse it right now? Probably not since it's more than an hour.

Reply:
I've added it to my todo list's `sidequests` section for now